lib/puppet/functions/pulpcore/generate_header_content.rb

```diff
@@ -0,0 +1,20 @@
+Puppet::Functions.create_function(:'pulpcore::generate_header_content') do
```

Comment:
I suppose a name like `wrap_and_comment_header_content` would be more descriptive here. Open to suggestions

Also trying to determine what would be the correct place to open a PR with this function upstream. Perhaps it makes the most sense in the Concat module?